src/main/kotlin/com/exactpro/th2/ws/client/Main.kt

```diff
@@ -133,9 +138,15 @@ fun run(
         handler,
         onMessage,
         onEvent
-    ).apply { registerResource("client", ::close) }
+    ).apply { registerResource("client", ::stop) }
+
+    val controller = ClientController(client).apply { registerResource("controller", ::close) }
 
     val listener = MessageListener<MessageGroupBatch> { _, message ->
+        if (!controller.isRunning) { // should we reschedule stop if service is already running?
+            controller.start(settings.autoStopAfter)
+        }
```

Comment:
We can face the problem client stops after this code and we hung up on the awaitSocket method